community/modules/compute/schedmd-slurm-gcp-v5-node-group/main.tf

```diff
@@ -23,10 +23,11 @@ locals {
 
   # Handle VM image format from 2 sources, prioritize source_image* variables
   # over instance_image
-  source_image_input_used = var.source_image != "" || var.source_image_family != "" || var.source_image_project != ""
-  source_image            = local.source_image_input_used ? var.source_image : lookup(var.instance_image, "name", "")
-  source_image_family     = local.source_image_input_used ? var.source_image_family : lookup(var.instance_image, "family", "")
-  source_image_project    = local.source_image_input_used ? var.source_image_project : lookup(var.instance_image, "project", "")
+  source_image_input_used         = var.source_image != "" || var.source_image_family != "" || var.source_image_project != ""
+  source_image                    = local.source_image_input_used ? var.source_image : var.instance_image.name
+  source_image_family             = local.source_image_input_used ? var.source_image_family : var.instance_image.family
```

Comment:
The original code was performing lookups on `instance_image` for name and family (and project). What happens in the case of `var.instance_image.name` if name is not provided (it is optional). I suspect it will fail here or null will be returned and fail lower in the stack.

My gut is that the lookups were there for a reason and we will be safer to keep them.

Reply:
Added lookups.